src/DIRAC/Core/Utilities/JEncode.py

```diff
@@ -88,6 +90,21 @@ class DJSONEncoder(json.JSONEncoder):
     tuple, datetime, and any object inheriting from JSerializable
     """
 
+    # TODO: remove it in 8.0
+    def encode(self, obj):
+        """This method for backward compatability with python 2 code"""
+        if six.PY2:
+            # Json can't distinguish bytes or str in python 2, so it's trying to encode with utf-8/ascii
+            try:
+                # Let's leave this conversion for backward compatibility, although
+                # it does not guarantee the return of the same data when decoding
+                return super(DJSONEncoder, self).encode(obj)
+            except UnicodeDecodeError as e:
+                # If the encoding type does not match we have an exception
+                # that json does not try to fix using default method
+                obj = self.default(obj)  # Let's use base64 in this case
+        return super(DJSONEncoder, self).encode(obj)
+
```

Comment:
Well, the current implementation just guarantees the correct operation of `JEncode` with bytes on Python 3, in all other cases `JEncode` tries to do everything possible to increase the stability of the transfer and at the same time provide backward compatibility.

Clearly banning the use of bytes in python 2 will probably be harder than trying to pass them, because in python 2 the line between `str` and `bytes` is very blurred

Reply:
> Which means it will fail in a seemingly non-deterministic way depending on the content being sent

correctly, which in principle does not change the current situation, just a little "smoothes the corners" (in the case of complex bytes, we can be sure that these are bytes and pass them as bytes). (python 2)